### ISSUE TITLE
New setup wizard

### ISSUE THREAD

**Issue by BobrImperator** [blogic/ucentral-pwa#6]
After logging in on a fresh firmware, a setup wizard needs to be run in order to create WiFi configuration.

![NewSetup_WiFiName](https://user-images.githubusercontent.com/15948633/118635612-0e8cf280-b7d4-11eb-94b8-d766c8cc6e9b.png)
First step - enter WiFi name
https://share.goabstract.com/b77a898c-b1e9-447b-b657-d8d639601ad5
Second step - enter WiFi password
https://share.goabstract.com/5f72174b-21a7-4587-ba3a-8a7d3acce511
Third step will be blank for now and resolved in another task.
https://share.goabstract.com/d4169290-b44e-49fb-b441-a678a4d56bc9?mode=design&sha=c609991f3db28c212d5b7a820c3a81423b3681a6

Mesh network setup is a multi-step form with a header, description, step indicators, button and a labeled input on each step. As well as a view with a QR code, sharing option and a link once the setup is finished.

On first step a wifi name is entered, then on second step a password, once wizard starts setup it will show current setup step that's being processed, when succeeded a QR code is shown.

### Questions
- [x] When a user finishes the setup and settings are being applied to a device, how will we know the device is already set up, or is still in progress?
We’re thinking that long polling of some sorts, is there an endpoint that we could send requests to?

    > It will return a 200 on success and 404 if anything failed
once you get the 200, the controller has passed all information down to the devices. these will take a little to configure and there is no real way we can tell if they are done.

    - So we’ve passed the info to the device, we received a `200` response which means we can proceed and generate a QR code for it.

reference qr code generator https://qifi.org/

### TODOS
- [x] Create 'new setup' route
- [x] NewSetup component
- [x] Multi Step manager component
We will use `ember-steps` addon to manage the wizard
- [x] Create step indicators 
- [x] Connect to Auth Flow #4 in order to redirect to `setup` for not yet existing configuration
   - [x] handle situation where device is already configured
- [x] Create `currentDevice` service with `isConfigured` property in order to determine to either redirect to dashboard or new setup.
- [x] When on the last step, if a request passes, show 'setting up' screen, if not, then go back to the first step in the setup and show 'Couldn't setup mesh network' message.

**Comment by marcoow** [comment]
> Could use ember-steps addon to manage the wizard

I'd suggest to decide on whether we want to use that or not before we begin implementation of this.

Also, I'd recommend leaving all of what's shown within the third step of the wizard out of this and handling that in another task. That would make this much more manageable and answer the above questions.

**Comment by BobrImperator** [comment]
@marcoow I changed the wording, I think ember-steps has a good API and is alive.
Updated Todos to match what you said.

**Comment by marcoow** [comment]
One important thing to consider here is that during the last step of the wizard the router will reboot and we will lose connection. We'll need to poll or so to check when it eventually comes back online.
